### PR TITLE
libc/picolibc: Picolibc fixes

### PR DESCRIPTION
**Pull request by keith-packard**
These are a handful of fixes required to use picolibc as the default C library

### CODE REVIEW

**Review by joerchan — COMMENTED** [review]
Approving TF-M sample change

**Review by joerchan — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

File: samples/tfm_integration/tfm_psa_test/src/main.c [hunk 1]
```diff
@@ -9,7 +9,6 @@
 /* Run the PSA test suite */
 void psa_test(void);
 
-__attribute__((noreturn))
 int main(void)
```
Comment (dcpleung) [inline samples/tfm_integration/tfm_psa_test/src/main.c:12]: Does this need to return something even with the for loop in the end? Some toolchains might complain that there is no return statement.

Comment (keith-packard) [reply samples/tfm_integration/tfm_psa_test/src/main.c:12]: You have the choice between a warning about a missing return and a warning for unreachable code. It looks like gcc (at least) is happy with the code as-is.

Comment (microbuilder) [reply samples/tfm_integration/tfm_psa_test/src/main.c:12]: @dcpleung Any further feedback on this one?

Comment (dcpleung) [reply samples/tfm_integration/tfm_psa_test/src/main.c:12]: No. As long as the compiler is happy, I am fine with this.

**Comment by keith-packard** [comment]
I moved the modules/audio/sof fix out of this PR as that didn't fit the other changes here which are all minor fixes to samples.

**Review by dcpleung — APPROVED** [review]

**Review by microbuilder — APPROVED** [review]

**Pull request merged**